lib/universal_id/portable_hash.rb

```diff
@@ -39,7 +39,9 @@ def hydrate(hash)
 
     def deep_dehydrate(value, options:)
       value = if implements_gid?(value)
-        implements_gid?(value) ? value.to_gid_param : value
+        value.to_gid_param
+      elsif defined?(ActiveRecord) && (value.is_a?(ActiveRecord::Associations::CollectionProxy) || value.is_a?(ActiveRecord::Relation))
+        deep_dehydrate(value.to_a, options: options)
```

Comment:
This will force a load and hydration of all records the query matches. If we choose to continue with this strategy, I'd recommend that use update to only select the primary key column which should be all we need to create the gid.

Ideally I'd like to see us use a strategy similar to how associations manage cache keys. Namely that we extend ActiveRecord/Arel associations to support GlobalID by leveraging something like `to_sql` etc.

Reply:
You are correct, there's a lot to optimize.

This is a first pass on it, and the good new is that it works 🎉

Reply:
Updated this to use the `primary_key` only.